tests/tracer/test_span.py

```diff
@@ -479,6 +480,29 @@ def test_span_unicode_set_tag():
     span._set_str_tag(u"😐", u"😌")
 
 
+@pytest.mark.skipif(sys.version_info.major != 2, reason="This test only applies Python 2")
+@mock.patch("ddtrace.span.log")
+def test_span_binary_unicode_set_tag(span_log):
+    span = Span(None, None)
+    span.set_tag("key", "🤔")
+    span._set_str_tag("key_str", "🤔")
+    # only span.set_tag() will fail
```

Comment:
is there any reason not to use `ensure_text` also in `set_tag`?

Reply:
We could though in that case we'd need to first get the string representation of the value object if its not a `six.binary_type` or `six.text_type`. I wanted to keep this PR to just `_set_str_tag` but if this works for this case (which is broken) I'd do the same for `set_tag` (which is less broken but still doesn't properly handle these binary and byte string cases).

Reply:
I'm actually confused why `set_tag` would fail and `_set_str_tag`?

Reply:
By "fail" I mean that an exception will be raised and handled in the case of `set_tag`:

```
>>> six.text_type("🤔")
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
UnicodeDecodeError: 'ascii' codec can't decode byte 0xf0 in position 0: ordinal not in range(128)
>>> six.ensure_text("🤔")
u'\U0001f914'
```

Reply:
Oh yeah. I guess we really should be using `ensure_text` in `set_tag`. Using `text_type` sounds really like a bad idea. 😢